pinot-common/src/main/java/com/linkedin/pinot/common/protocols/SegmentFinalProtocol.java

```diff
@@ -0,0 +1,165 @@
+ * is set (in the table configuration), it sends a SegmentConsumedRequest to the controller (leader).  The
+ * controller may respond with a HOLD, asking the server to NOT consume any new kafka messages, but get back
+ * to the controller after a while.
+ *
+ * Meanwhile, the controller co-ordinates the SegmentConsumedRequest messages from replicas and selects a server
+ * to commit the segment. The server uses SegmentCommitRequest message, in which it also posts the completed
+ * segment to the controller.
+ *
+ * The controller may respond with a failure for this commit message, but on success, the controller changes the
+ * segment state to ONLINE in idealstate, and adds new CONSUMING segments as well.
+ *
+ * For details see the design document
+ * https://github.com/linkedin/pinot/wiki/Low-level-kafka-consumers
+ */
+public class SegmentFinalProtocol {
+  /**
+   * MAX_HOLD_TIME_MS is the maximum time (msecs) for which a server will be in HOLDING state, after which it will
+   * send in a SegmentConsumedRequest with its current offset in kafka.
+   */
+  public static final long MAX_HOLD_TIME_MS = 3000;
+  /**
+   * MAX_SEGMENT_COMMIT_TIME_MS is the longest time (msecs) a server will take to complete building a segment and committing
+   * it  (via a SegmentCommit message) after the server has been notified that it is the committer.
+   */
+  public static final long MAX_SEGMENT_COMMIT_TIME_MS = 15000;
+
+  public enum ControllerResponseStatus {
+    NOT_SENT, // Never used by the controller. Only used locally by the server when it is not able to send a request
+    COMMIT,
+    HOLD,
+    CATCHUP,
+    DISCARD,
+    KEEP,
+    NOT_LEADER,
+    FAILED,
+    COMMIT_SUCCESS,
+  }
+
+  public static final String STATUS_KEY = "status";
+  public static final String OFFSET_KEY = "offset";
+
+  public static final String MSG_TYPE_CONSUMED = "segmentConsumed";
+  public static final String MSG_TYPE_COMMMIT = "segmentCommit";
+
+  public static final String PARAM_SEGMENT_NAME = "name";
+  public static final String PARAM_OFFSET = "offset";
+  public static final String PARAM_INSTANCE_ID = "instance";
+
+  // Canned responses
+  public static final Response RESP_NOT_LEADER = new Response(ControllerResponseStatus.NOT_LEADER, -1L);
+  public static final Response RESP_FAILED = new Response(ControllerResponseStatus.FAILED, -1L);
+  public static final Response RESP_DISCARD = new Response(ControllerResponseStatus.DISCARD, -1L);
+  public static final Response RESP_COMMIT_SUCCESS = new Response(ControllerResponseStatus.COMMIT_SUCCESS, -1L);
+
+  public static abstract class Request {
+    final String _segmentName;
+    final long _offset;
+    final String _instanceId;
+
+    public Request(String segmentName, long offset, String instanceId) {
+      _segmentName = segmentName;
+      _instanceId = instanceId;
+      _offset = offset;
+    }
+
+    public abstract String getUrl(String hostPort);
+
+    protected String getUri(String method) {
```

Comment:
It refers to one of MSG_TYPE_CONSUMED or MSG_TYPE_COMMIT

I will change it to msgType (or pathComponent)